spec-tests/src/test/java/org/candlepin/spec/CloudRegistrationSpecTest.java

```diff
@@ -796,4 +801,59 @@ private static Stream<Arguments> tokenVariation() throws ApiException {
             Arguments.of(owner.getKey(), null, "test_signature"),
             Arguments.of(owner.getKey(), "test-type", null));
     }
+
+    @Nested
+    @Isolated
+    @Execution(ExecutionMode.SAME_THREAD)
+    class WithScheduler {
+
+        @AfterEach
+        void tearDown() {
+            ApiClients.admin().jobs().setSchedulerStatus(true);
+        }
+
+        @Test
+        public void shouldCancelCloudOrgSetUpJobs() {
+            ApiClients.admin().jobs().setSchedulerStatus(false);
+            ApiClient adminClient = ApiClients.admin();
+            ProductDTO productDTO = Products.random();
+
+            String accountId = StringUtil.random("cloud-account-id-");
+            String instanceId = StringUtil.random("cloud-instance-id-");
+            String offerId = StringUtil.random("cloud-offer-");
+
+            adminClient.hosted().createProduct(productDTO);
+            adminClient.hosted().associateProductIdsToCloudOffer(offerId, List.of(productDTO.getId()));
+
+            OffsetDateTime timeBeforeJobStarts = OffsetDateTime.now();
+            // Because MySQL/Mariadb versions before 5.6.4 truncate milliseconds, lets remove a couple seconds
+            // to ensure we will not miss the job we need in the job query results later on.
+            timeBeforeJobStarts = timeBeforeJobStarts.minusSeconds(2);
+
+            CloudAuthenticationResultDTO result = ApiClients.noAuth().cloudAuthorization()
+                .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+
+            assertTokenType(ApiClient.MAPPER, result.getToken(), ANON_TOKEN_TYPE);
+
+            List<AsyncJobStatusDTO> jobs = adminClient.jobs().listJobStatuses(null,
+                Set.of("CloudAccountOrgSetupJob"), Set.of("CREATED"), null, Set.of("Anonymous"), null, null,
+                timeBeforeJobStarts, null, null, null, null, null);
+
+            assertThat(jobs).hasSize(1);
+            String beforeJobId = jobs.get(0).getId();
+            assertThatJob(jobs.get(0)).isCreated();
+
+            adminClient.cloudAuthorization().cancelCloudAccountJobs(accountId);
+
+            jobs = adminClient.jobs().listJobStatuses(null,
+                Set.of("CloudAccountOrgSetupJob"), Set.of("CANCELED"), null, Set.of("Anonymous"), null, null,
+                timeBeforeJobStarts, null, null, null, null, null);
```

Comment:
`cancelCloudAccountJobs` should now be returning a list of cancelled job ids, which you can use to pass into `listJobStatuses`, instead of all these other variables now: 
`listJobStatuses(Set.copyOf(returnList), null, null, null...)`

Reply:
done